src/main/webapp/app/core/course/manage/course-management-container/course-management-container.component.ts

```diff
@@ -0,0 +1,353 @@
+            this.updateRecentlyAccessedCourses().catch();
+        });
+    }
+
+    loadCourse(): Observable<void> {
+        return this.courseManagementService.findOneForDashboard(this.courseId()!).pipe(
+            map((res: HttpResponse<Course>) => {
+                if (res.body) {
+                    this.course.set(res.body);
+                }
+                this.setupConversationService();
+            }),
+        );
+    }
+
+    protected getHasSidebar(): boolean {
+        return this.communicationRouteLoaded();
+    }
+
+    /** Navigate to a new Course */
+    switchCourse(course: Course) {
+        const url = ['course-management', course.id];
+        this.router.navigate(url).then(() => {
+            this.handleCourseIdChange(course.id!);
+        });
+    }
+
+    protected handleComponentActivation(componentRef: any): void {
+        if (
+            componentRef instanceof CourseDetailComponent ||
+            componentRef instanceof CourseManagementExercisesComponent ||
+            componentRef instanceof ExamManagementComponent ||
+            componentRef instanceof LectureComponent ||
+            componentRef instanceof CourseManagementStatisticsComponent ||
+            componentRef instanceof CourseConversationsComponent ||
+            componentRef instanceof TutorialGroupsChecklistComponent ||
+            componentRef instanceof CompetencyManagementComponent ||
+            componentRef instanceof LearningPathInstructorPageComponent ||
+            componentRef instanceof AssessmentDashboardComponent ||
+            componentRef instanceof CourseScoresComponent ||
+            componentRef instanceof FaqComponent ||
+            componentRef instanceof BuildQueueComponent
+        ) {
+            this.activatedComponentReference.set(componentRef);
+        }
+        if (this.activatedComponentReference() instanceof CourseConversationsComponent) {
+            const childRouteComponent = this.activatedComponentReference() as CourseConversationsComponent;
+            this.isSidebarCollapsed.set(childRouteComponent?.isCollapsed ?? false);
+        }
+    }
+
+    handleToggleSidebar(): void {
+        if (!this.activatedComponentReference() || !(this.activatedComponentReference() instanceof CourseConversationsComponent)) {
+            return;
+        }
+        const childRouteComponent = this.activatedComponentReference() as CourseConversationsComponent;
+        childRouteComponent.toggleSidebar();
+        this.isSidebarCollapsed.set(childRouteComponent.isCollapsed);
+    }
+
+    override getSidebarItems(): SidebarItem[] {
+        const sidebarItems: SidebarItem[] = [];
+        const currentCourse = this.course();
+
+        sidebarItems.push(...this.sidebarItemService.getManagementDefaultItems(this.courseId()));
+        if (currentCourse?.isAtLeastEditor) {
+            sidebarItems.splice(3, 0, this.sidebarItemService.getLecturesItem(this.courseId()));
+        }
```

Comment:
We seem to need `curentCourse` for all following values, can we please use return based programming instead?

```
if (!currentCourse) {
    return;
}
```

Reply:
(no super strong opinion)